corehq/apps/api/resources/v0_5.py

```diff
@@ -244,6 +244,11 @@ def _update(self, bundle):
                             return False
                     bundle.obj.set_password(bundle.data.get("password"))
                     should_save = True
+                elif key == 'user_data':
+                    try:
+                        bundle.obj.metadata = value
```

Comment:
This is a good point, I've been mulling it over since Friday. I'm inclined to leave it as-is, to follow precedent and because I don't have a good alternative. Projects that use profiles and the API will be subject to errors if they update user_data in a way that conflicts with the profile, so it doesn't seem right to not let them access the profile.

There are other system keys in user data, including the user's location, that are similarly undocumented and have the same risk, which makes me think people either don't run into it or have learned to pull the user_data and then add/remove keys, rather than wholesale replacing it. I think I'll add that as a best practice to the API docs.

Reply:
I think this is sound rationale. I saw your docs update - I think it might be worth mentioning in there the specific keys that can be dangerous to overwrite, or give a pointer to how to "pull" the latest user data.

Reply:
👍  updated